### PR TITLE
SPOI-12270 Adding malhar jdbc operators with changes to handle empty …

### PR DESCRIPTION
**Pull request by deepak-narkhede**
…string for where clause.

### CODE REVIEW

**Comment by deepak-narkhede** [comment]
@yogidevendra. Please review.

File: app-templates/database-to-database-sync/pom.xml [hunk 1]
```diff
@@ -263,7 +260,7 @@
 
   <dependencies>
     <!-- add your dependencies here -->
-    <dependency>
+  <!--  <dependency>
```
Comment (yogidevendra) [inline app-templates/database-to-database-sync/pom.xml]: why this commented? should we remove this?

File: app-templates/database-to-database-sync/src/main/java/com/datatorrent/apps/Application.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ public void populateDAG(DAG dag, Configuration conf)
     /*
      * Jdbc Input and Output operators.
      */
-    JdbcPOJOPollInputOperator jdbcInputOperator = dag.addOperator("JdbcInput", new JdbcPOJOPollInputOperator());
+    JdbcPOJOPollInputOperatorMetrics jdbcInputOperator = dag.addOperator("JdbcInput", new JdbcPOJOPollInputOperatorMetrics());
```
Comment (yogidevendra) [inline app-templates/database-to-database-sync/src/main/java/com/datatorrent/apps/Application.java]: Why do we have two different implementations in moodi?
What is difference in JdbcPOJOPollInputOperator Vs JdbcPOJOPollInputOperatorMetrics

**Comment by yogidevendra** [comment]
Given 2 suggestions for changes.

[optional] It would be nice if you can split this into two commits.
1. With copying files from malhar to moodi.
2. Specific changes for the fix.

File: app-templates/database-to-database-sync/pom.xml (hunk 1 above)
Comment (deepak-narkhede) [inline app-templates/database-to-database-sync/pom.xml]: Was testing it. Fixed it.

File: app-templates/database-to-database-sync/src/main/java/com/datatorrent/apps/Application.java (hunk 2 above)
Comment (deepak-narkhede) [inline app-templates/database-to-database-sync/src/main/java/com/datatorrent/apps/Application.java]: JdbcPOJOPollInputOperatorMetrics is extension for JdbcPOJOPollInputOperator malhar operator. It has metrics included in it.If the changes would have been merged, maintaining would have been difficult.

Comment (yogidevendra) [inline app-templates/database-to-database-sync/src/main/java/com/datatorrent/apps/Application.java]: Finally, we want single operator with all enhancements and bug fixes. 
We cannot say that enhancements are in operator and bug fix is in another operator.

File: operators/library/src/main/java/com/datatorrent/moodi/lib/io/db/AbstractJdbcPollInputOperator.java [hunk 3]
```diff
@@ -0,0 +1,750 @@
...
+  private int partitionCount = 1;
+  private int batchSize = DEFAULT_BATCH_SIZE;
+
+  @NotNull
+  private String tableName;
+  @NotNull
+  private String columnsExpression;
+  @NotNull
+  private String key;
+  private String whereCondition = null;
```
Comment (yogidevendra) [inline operators/library/src/main/java/com/datatorrent/moodi/lib/io/db/AbstractJdbcPollInputOperator.java]: change this default to empty string.

Comment (deepak-narkhede) [inline operators/library/src/main/java/com/datatorrent/moodi/lib/io/db/AbstractJdbcPollInputOperator.java]: Done.

File: app-templates/database-to-database-sync/src/main/java/com/datatorrent/apps/Application.java (hunk 2 above)
Comment (deepak-narkhede) [inline app-templates/database-to-database-sync/src/main/java/com/datatorrent/apps/Application.java]: Done.

File: operators/library/src/main/java/com/datatorrent/moodi/lib/io/db/AbstractJdbcPollInputOperator.java (hunk 3 above)
Comment (yogidevendra) [inline operators/library/src/main/java/com/datatorrent/moodi/lib/io/db/AbstractJdbcPollInputOperator.java]: Let me know once changes are pushed.

**Comment by deepak-narkhede** [comment]
@yogidevendra. Incorporated the changes. Please review and merge.

**Pull request merged**